carts/models.py

```diff
@@ -0,0 +1,106 @@
+"""
+Models for cart app
+"""
+from decimal import Decimal
+
+from django.core.validators import MinValueValidator
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from carts.constants import CART_STATUS, OPEN
+from products.models import AuditTimeStamp, Product
+from users.models import User
+
+
+class Cart(AuditTimeStamp):
+    """
+    Model class that represents Cart
+    """
+
+    status = models.CharField(max_length=40, choices=CART_STATUS, default=OPEN)
+    user = models.ForeignKey(User, on_delete=models.PROTECT, related_name="cart")
+    total_bill = models.DecimalField(
+        _("Total Payable"),
+        max_digits=20,
+        validators=(MinValueValidator(1),),
+        decimal_places=2,
+        default=Decimal(0),
+    )
+
+    def update_bill(self):
+        """
+        Updates the bill by traversing all the cart items
+        Returns:
+            None
+        """
+        # pylint: disable=no-member
+        total = Decimal(0)
+        for item in self.cart_items.all():
+            total += item.get_total()
+        self.total_bill = Decimal(total)
+        self.save()
+
+    def __str__(self):
+        return f"{self.user.name}'s Cart"
+
+
+class CartItem(AuditTimeStamp):
+    """
+    Model class that represents a CartItem
+    """
+
+    product = models.ForeignKey(Product, on_delete=models.CASCADE, related_name="+")
+    cart = models.ForeignKey(Cart, on_delete=models.CASCADE, related_name="cart_items")
+    quantity = models.PositiveIntegerField(
```

Comment:
Just wanted to add a note on something i think currently happening:

Technically we should not have any associated CartItem to a cart with quality 0. 0 quantity will mean we should delete the item from the cart.

Reply:
There is no such association. A cart item can not have `quantity` of `0` because there `MinValueValidator(1)` ensures it. Is this what you are trying to convey? Or did i misunderstand your comment. Please let me know

Reply:
I'm sorry, I think I wasn't clear enough let me explain more:

1- A user can add/delete items from a cart. Correct?
2 - Let's say a user added `ItemA` in his cart and the API will eventually add one `CartItem` entry with quantity 1 in the cart
3- What if the user now deletes `ItemA` from his cart? Would the cart still have `CartItem` with quantity 0 or the `CartItem` will completely be deleted?

Does the above make more sense now?

Reply:
The `CartItem` will completely be deleted from the table